src/app/gps/qgsappgpsconnection.cpp

```diff
@@ -176,12 +178,18 @@ void QgsAppGpsConnection::disconnectGps()
 
 void QgsAppGpsConnection::onTimeOut()
 {
+  std::unique_ptr< QgsGpsConnection > oldConnection( mConnection );
```

Comment:
This is duplicate logic from disconnectGps(). How about simplifying this to something like:

```
void QgsAppGpsConnection::onTimeOut()
{
   disconnectGps();

  emit connectionTimedOut();
  QgisApp::instance()->statusBarIface()->clearMessage();
  showGpsConnectFailureWarning( tr( "TIMEOUT - Failed to connect to GPS device." ) );
}
```